cmd/wstail/tail.go

```diff
@@ -42,9 +42,15 @@ func writer(ws *websocket.Conn, filter filter) {
 	log.Infof("active filter: %v (%v)", filter, matcher)
 	defer ws.Close()
 
-	tailCmd := exec.Command("tail", fmt.Sprintf("-n%v", tailHistory), "-f", filePath)
+	var err error
+	var history io.ReadCloser
+	tailCmd := exec.Command("tail", "-f", filePath)
 	commands := []*exec.Cmd{tailCmd}
 	if matcher != "" {
+		history, err = snapshot(matcher, filePath, tailHistory)
+		if err != nil {
+			log.Errorf("failed to obtain %v lines of history for %v: %v", tailHistory, matcher, trace.DebugReport(err))
```

Comment:
Showing history is not vital - the tail aborts only if the actual tailing fails. With no history obtained, there's simply not historical output.